Reflection_Engine/Modify/SortExtensionMethods.cs

```diff
@@ -42,11 +42,8 @@ public static partial class Modify
         [Output("metods", "Sorted methods")]
         public static List<MethodInfo> SortExtensionMethods(this IEnumerable<MethodInfo> methods, Type type)
         {
-            //TODO: Sort methods based on closeness to the type, not just exact vs non exact.
-            //Example A : B and B : C
-            //Then if the type is A and list of methods contain one method with first parameter matching each type, then the list should be
-            //Sorted so that the method with A comes first followed by B and lastly C.
-            return methods.OrderBy(x => x.GetParameters().FirstOrDefault()?.ParameterType == type ? 0 : 1).ToList();
+            IEnumerable<MethodInfo> typeMethods = methods.Where(x => x.GetParameters()[0].ParameterType.IsAssignableFrom(type));
+            return methods.OrderBy(x => methods.Count(y => x.GetParameters()[0].ParameterType.IsAssignableFrom(y.GetParameters()[0].ParameterType))).ToList();
```

Comment:
Not sure if I agree with this change. The purpose of this method is to say that say `Geometry(Bar bar)` should be prioritised over `Geometry(IElement1d element)`. That is, exact match before any interface level.

By using `IsAssignableFrom` it feels like both of those cases will be returning the same match...

Reply:
Not really, the methods are ordered by the number of target types their target type is assignable from, so for example:
```
MyMethod(Line)
MyMethod(ICurve)
MyMethod(IGeometry)
```

`MyMethod(Line)` will score 1 (`typeof(Line)` is assignable from `typeof(Line)` only), the others 2 and 3 respectively.

In your case, `Geometry(Bar bar)` should score 1, `Geometry(IElement1d element)` - 2. So the former will be prioritised.

Does that make sense or am I missing a point?